src/main.c

```diff
@@ -1140,9 +1177,18 @@ static int on_config_listen(h2o_configurator_command_t *cmd, h2o_configurator_co
                             freeaddrinfo(res);
                             return -1;
                         }
+#if defined(__linux__) && defined(SO_REUSEPORT)
+                        if (so_reuseport) {
+                            int flag = 1;
+                            if (setsockopt(fd, SOL_SOCKET, SO_REUSEPORT, &flag, sizeof(flag)) != 0) {
```

Comment:
Does setting the flag here work correctly?

`man 7 socket` states that "this option must be set on each socket (including the first socket) prior to calling bind(2)". However it seems that we are grabbing a socket already being bound (by server-starter) here.

FWIW, you can test the behavior by passing `--mode=master` or `--mode=daemon` to `h2o`.

Reply:
sorry for missing the point: 
"this option must be set on each socket (including the first socket) prior to calling bind(2)"

`man 7 socket` on my system missing this information...